File: jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/SignedIntent.kt
package com.radixdlt.sargon.extensions

import com.radixdlt.sargon.SignedIntent
import com.radixdlt.sargon.SignedTransactionIntentHash
import com.radixdlt.sargon.signedIntentHash

fun SignedIntent.hash(): SignedTransactionIntentHash = signedIntentHash(signedIntent = this)

File: crates/uniffi/uniffi_SPLIT_ME/src/wrapped_radix_engine_toolkit/low_level/signed_intent.rs
use crate::prelude::*;
use sargon::SignedIntent as InternalSignedIntent;

#[derive(Clone, PartialEq, Eq, InternalConversion, uniffi::Record)]
pub struct SignedIntent {
    intent: TransactionIntent,
    pub intent_signatures: IntentSignatures,
}

#[uniffi::export]
pub fn signed_intent_hash(
    signed_intent: &SignedIntent,
) -> SignedTransactionIntentHash {
    signed_intent.into_internal().hash().into()
}

#[uniffi::export]
pub fn new_signed_intent_sample() -> SignedIntent {
    InternalSignedIntent::sample().into()
}

#[uniffi::export]
pub fn new_signed_intent_sample_other() -> SignedIntent {
    InternalSignedIntent::sample_other().into()
}
